### PR TITLE
[sw/crt] Fix CRT code clobbering the first word of .bss

### PR DESCRIPTION
**Pull request by mcy**
When this assembly was written, an incorrect assumption was made that
the _*_end pointers pointed to the final word in a section, when in
reality, they point one past. Due to how the linker laid out .data and
.bss, .data initialization would clobber the first word of .bss.

### CODE REVIEW

**Comment by mcy** [comment]
@lenary Do you know if the behavior of `symbol = .;` differs between GCC and Clang? AFAICT (from testing) it doesn't... but the toolchain switchover makes me a bit nervous.

Also, I should probably write a test for this change before merging it... not sure what that should look like though.

File: sw/device/boot_rom/rom_crt.S [hunk 1]
```diff
@@ -82,21 +82,21 @@ _start:
 bss_zero_loop:
   sw    zero, 0(t0)
   addi  t0, t0, 0x4
-  ble   t0, t1, bss_zero_loop
+  blt   t0, t1, bss_zero_loop
```
Comment (lenary) [inline sw/device/boot_rom/rom_crt.S:85]: 🥪🥓🥬🍅🥪

**Comment by lenary** [comment]
Oh, by "would it be safer if", i mean in addition to these changes, not instead of.

**Comment by mcy** [comment]
Added a test. It can catch the regression I fixed, but it can also pass spuriously... so *shrug*

File: sw/device/tests/crt_test.c [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  CHECK(data_end_addr % sizeof(uint32_t) == 0,
+        "_data_end not word-aligned: 0x%08x", data_end_addr);
+  CHECK(data_init_start_addr % sizeof(uint32_t) == 0,
+        "_data_init_start not word-aligned: 0x%08x", data_init_start_addr);
+
+  CHECK(bad_bss_index == -1, "found non-zero .bss byte at *0x%08x == 0x%02x",
+        bss_start_addr + bad_bss_index, (uint32_t)bss[bad_bss_index]);
+  CHECK(bad_data_index == -1,
+        "found bad .data byte at *0x%08x == 0x%02x, expected 0x%02x",
+        data_start_addr + bad_data_index, (uint32_t)data[bad_data_index]);
```
Comment (lenary) [inline sw/device/tests/crt_test.c:107]:
```suggestion
        data_start_addr + bad_data_index, (uint32_t)data_init[bad_data_index]);
```

**Pull request merged**